presto-parquet/src/test/java/com/facebook/presto/parquet/TestParquetFileStats.java

```diff
@@ -0,0 +1,206 @@
+            assertTrue(maxBinary.compareTo(Binary.fromString(testString)) >= 0, String.format("Binary max: %s,str: %s", maxBinary.toStringUsingUTF8(), testString));
+        }
+        String max = maxBinary.toStringUsingUTF8();
+        String min = minBinary.toStringUsingUTF8();
+        // thrown assert-error when using parquet-mr prior 1.10.0 (PARQUET-1025),
+        // effects TupleDomainParquetPredicate.getDomain
+        // means
+        //    not-contains-empty-string in parquet-blocks the condition minSlice.compareTo(maxSlice) always true
+        //    contains-empty-string be false, but only accept empty-string and min-string will filter more blocks than expected
+        assertTrue(Slices.utf8Slice(max).compareTo(Slices.utf8Slice(min)) >= 0, "Binary max not gather than min");
+        // calculate min,max value for slice
+        Slice maxSlice = null;
+        Slice minSlice = null;
+        for (String testString : strings) {
+            Slice slice = Slices.utf8Slice(testString);
+            if (maxSlice == null || maxSlice.compareTo(slice) < 0) {
+                maxSlice = slice;
+            }
+            if (minSlice == null || minSlice.compareTo(slice) > 0) {
+                minSlice = slice;
+            }
+        }
+        // thrown assert-error when using parquet-mr prior 1.10.0 (PARQUET-1025)
+        assertEquals(minSlice, Slices.utf8Slice(min), "Min-binary and Min-slice mismatch");
+        assertEquals(maxSlice, Slices.utf8Slice(max), "Max-binary and Max-slice mismatch");
+    }
+
+    @Test
+    public void testParquetFile()
+            throws Exception
+    {
+        testParquetFile(testStrings, false);
+        testParquetFile(testStrings, true);
+    }
+
+    public void testParquetFile(List<String> strings, boolean containEmptyString)
+            throws Exception
+    {
+        if (containEmptyString) {
+            strings.add("");
+        }
+        Configuration conf = new Configuration();
+        Path root = new Path("target/tests/TestParquetWriter/");
+        enforceEmptyDir(conf, root);
+        MessageType schema = parseMessageType(
+                "message test { "
+                        + "required int32 int32_field; "
+                        + "required binary binary_field; "
+                        + "required fixed_len_byte_array(3) flba_field; "
+                        + "} ");
+        GroupWriteSupport.setSchema(schema, conf);
+        SimpleGroupFactory f = new SimpleGroupFactory(schema);
+        Map<String, Encoding> expected = new HashMap<String, Encoding>();
+        expected.put("10-" + PARQUET_1_0, PLAIN_DICTIONARY);
+        expected.put("1000-" + PARQUET_1_0, PLAIN);
+        expected.put("10-" + PARQUET_2_0, RLE_DICTIONARY);
+        expected.put("1000-" + PARQUET_2_0, DELTA_BYTE_ARRAY);
+        for (int modulo : asList(10, 1000)) {
+            for (ParquetProperties.WriterVersion version : ParquetProperties.WriterVersion.values()) {
```

Comment:
The list of expected encodings is hardcoded and includes only `PARQUET_1_0` and `PARQUET_2_0` versions. However, this loop can potentially support more versions. If a new version is added to the enum, the test will break.

I suggest to either change the loop to loop over `PARQUET_1_0` and `PARQUET_2_0` versions explicitly or change the `expected` map to include values for all versions in the `ParquetProperties.WriterVersion` enum.